highlighter/src/lib.rs

```diff
@@ -273,53 +279,177 @@ impl Highlight {
+    MonokaiExtendedOrigin,
+    Nord,
+    OneHalfDark,
+    OneHalfLight,
     SolarizedDark,
-    Base16Mocha,
-    Base16Ocean,
-    Base16Eighties,
-    InspiredGitHub,
+    SolarizedLight,
+    SublimeSnazzy,
+    TwoDark,
+    VisualStudioDarkPlus,
+    Zenburn,
 }
 
 impl Theme {
     /// A static slice containing all the available themes.
     pub const ALL: &'static [Self] = &[
+        Self::Ansi,
+        Self::Base16,
+        Self::Base16EightiesDark,
+        Self::Base16MochaDark,
+        Self::Base16OceanDark,
+        Self::Base16OceanLight,
+        Self::Base16_256,
+        Self::ColdarkCold,
+        Self::ColdarkDark,
+        Self::DarkNeon,
+        Self::Dracula,
+        Self::Github,
+        Self::GruvboxDark,
+        Self::GruvboxLight,
+        Self::InspiredGithub,
+        Self::Leet,
+        Self::MonokaiExtended,
+        Self::MonokaiExtendedBright,
+        Self::MonokaiExtendedLight,
+        Self::MonokaiExtendedOrigin,
+        Self::Nord,
+        Self::OneHalfDark,
+        Self::OneHalfLight,
         Self::SolarizedDark,
-        Self::Base16Mocha,
-        Self::Base16Ocean,
-        Self::Base16Eighties,
-        Self::InspiredGitHub,
+        Self::SolarizedLight,
+        Self::SublimeSnazzy,
+        Self::TwoDark,
+        Self::VisualStudioDarkPlus,
+        Self::Zenburn,
     ];
 
     /// Returns `true` if the [`Theme`] is dark, and false otherwise.
     pub fn is_dark(self) -> bool {
         match self {
-            Self::SolarizedDark
-            | Self::Base16Mocha
-            | Self::Base16Ocean
-            | Self::Base16Eighties => true,
-            Self::InspiredGitHub => false,
+            Theme::Ansi => true,
+            Theme::Base16 => true,
+            Theme::Base16EightiesDark => true,
+            Theme::Base16MochaDark => true,
+            Theme::Base16OceanDark => false,
+            Theme::Base16OceanLight => false,
+            Theme::Base16_256 => true,
+            Theme::ColdarkCold => false,
+            Theme::ColdarkDark => true,
+            Theme::DarkNeon => true,
+            Theme::Dracula => false,
+            Theme::Github => false,
+            Theme::GruvboxDark => true,
+            Theme::GruvboxLight => false,
+            Theme::InspiredGithub => false,
+            Theme::Leet => true,
+            Theme::MonokaiExtended => true,
+            Theme::MonokaiExtendedBright => true,
+            Theme::MonokaiExtendedLight => false,
+            Theme::MonokaiExtendedOrigin => true,
+            Theme::Nord => true,
+            Theme::OneHalfDark => true,
+            Theme::OneHalfLight => false,
+            Theme::SolarizedDark => true,
+            Theme::SolarizedLight => false,
+            Theme::SublimeSnazzy => true,
+            Theme::TwoDark => true,
+            Theme::VisualStudioDarkPlus => true,
+            Theme::Zenburn => true,
         }
     }
 
     fn key(self) -> &'static str {
         match self {
```

Comment:
This function's implementation is the same as [`two_face::theme::EmbeddedThemeName::as_name()`'s](https://docs.rs/two-face/latest/two_face/theme/enum.EmbeddedThemeName.html#method.as_name). You could use `.as_name()` if there was a `impl From<Theme> for EmbeddedThemeName` which would avoid having them as "magic values" here :thinking:

Reply:
Okay thanks, I was following the code pattern before just adding some variants.

Reply:
After looking into this, I can't find a way to compare iced's highlighter themes to two face's embeddedtheme. Because you would have to go from iced's theme to two face and without this key code theres no way to do that. The constant at the type of the file is a lazythemeset but thats not what iced's setting type holds.

Reply:
If you have an idea reopen this.

Reply:
Tried out my idea with this commit https://github.com/CosmicHorrorDev/iced/commit/0045223ca5e337cdfe3a4684816244f996511f41